### PR TITLE
feat: add yard doc for dsl methods

### PR DESCRIPTION
**Pull request by Mahaveer1141**
## What does this change do?
- add yard doc for dsl methods

## Why is this change needed?
- need documentation for dsl methods.

## How were the changes done?
- installed `yard` gem.
- added yard doc comments on each method for dsl method

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ gem 'pundit'
 gem 'rake', '~> 12.0'
 gem 'rspec', '~> 3.0'
 gem 'slim'
+gem 'yard'
```
Comment (mikevic) [inline Gemfile:9]: @Mahaveer1141 - does this need to be added to the Gemfile? From what I understand it can be run externally.

**Review by mikevic — CHANGES_REQUESTED** [review]
@Mahaveer1141 - let us configure this to build with Github Pages too? Let me know if any additional permissions are required.

File: Gemfile (hunk 1 above)
Comment (Mahaveer1141) [reply Gemfile:9]: Yeah, it doesn't need to be here, we need yard executable which can be installed externally also. But users need to install it externally then?

Comment (mikevic) [reply Gemfile:9]: @Mahaveer1141 - bro, is there a way to add it as a development dependency?

Comment (Mahaveer1141) [reply Gemfile:9]: we can but I think, if we are going to use github action to deploy this, then we can get this executable from bundle install only?

Comment (mikevic) [reply Gemfile:9]: @Mahaveer1141 - I'm not too sure either, I'll leave it to you to experiment.

**Pull request merged**